@core/cms/src/external-auth-integration/auth-providers/oauth2/slack/v2/slack.v2.service.ts

```diff
@@ -0,0 +1,109 @@
+import { Injectable, Inject, Logger, Res, HttpException, HttpStatus } from "@nestjs/common";
+import { HttpService } from "@nestjs/axios";
+import { EncryptionDecryptionService } from "~/encryption-decryption/encryption-decryption.service";
+import { IOAuth } from "~/external-auth-integration/auth-providers/oauth2/interface/ioauth.interface";
+import {
+  OAuth2StateProcessor,
+  exchangeCodeForToken,
+  verifyToken,
+  refreshToken,
+  createOAuth2Url,
+} from "~/external-auth-integration/utils";
+import { IOAuth2Config, IOAuth2State, TokenVerificationResponse } from "~/external-auth-integration/auth-providers/oauth2/@types";
+
+/**
+ * Service to handle Slack V2 OAuth2 authentication.
+ */
+@Injectable()
+export class SlackV2OAuth2Service implements IOAuth {
+  constructor(
+    @Inject("SlackV2OAuth2Config") private readonly config: IOAuth2Config,
+    private readonly encryptionDecryptionService: EncryptionDecryptionService,
+    private readonly httpService: HttpService,
+  ) {}
+
+  /**
+   * Builds and returns the authentication URL for Google OAuth.
+   * @returns {Promise<string>} The Google OAuth URL.
+   */
+  async authenticate(): Promise<string> {
+    const encodedState = await this.buildState();
+    // we are requesting user token, so we need to pass scopes to user_scope
+    const url = createOAuth2Url(this.config, {
+      state: encodedState,
+      scope: undefined
```

Comment:
why scope is undefined here?

Reply:
i thought it needs to not appear in the URL param, but that's unnecessary, just set the scope to an empty string and pass the scope to `user_scope`